src/huggingface_hub/hf_api.py

```diff
@@ -4772,6 +4772,92 @@ def delete_file(
             parent_commit=parent_commit,
         )
 
+    @validate_hf_hub_args
+    def delete_files_r(
+        self,
+        patterns: List[str],
+        repo_id: str,
+        *,
+        token: Union[bool, str, None] = None,
+        repo_type: Optional[str] = None,
+        revision: Optional[str] = None,
+        commit_message: Optional[str] = None,
+        commit_description: Optional[str] = None,
+        create_pr: Optional[bool] = None,
+        parent_commit: Optional[str] = None,
+    ) -> CommitInfo:
+        """
+        Recursively deletes files in a given repository based on Unix shell-style
+        file matching patterns.
+
+        Git based under the hood (server-side), meaning if you delete all files
+        from a directory, git untracks it. It works this way recursively
+        through the sub folders. Meaning a whole directory structure is untracked
+        by deleting their respectives files at all levels.
+
+        Also if one or many of the patterns from the `patterns` argument to
+        this function have a trailing `/` (as in `folder/`),
+        this function will interpret it as `/*` (as in `folder/*`) to comply with Unix
+        patterns (Unix shell standards don't match `folder/a` to `folder/` but
+        match it to `folder/*`) when deleting all files in the folders(s).
+        By the previous corollary, the folder(s) will disappear because all
+        their files are deleted.
+
+        Args:
+            path_in_repo (`str`):
+                Relative folder path in the repo, for example: `"checkpoints/1fec34a"`.
+            repo_id (`str`):
+                The repository from which the folder will be deleted, for example:
+                `"username/custom_transformers"`
```

Comment:
```suggestion
            repo_id (`str`):
                The repository from which the folder will be deleted, for example:
                `"username/custom_transformers"`
            delete_patterns (`List[str]`):
                List of files or folders to delete. Each string can either be a file path, a folder path or a Unix shell-style wildcard.
                E.g. `["file.txt", "folder/", "data/*.parquet"]`
```

`path_in_repo` is not a correct arg (`delete_patterns` instead) + should switch with repo_id since the change in args above.